packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -42,14 +40,14 @@ class XcodeProjectInterpreter {
   }
 
   XcodeProjectInterpreter._({
-    @required Platform platform,
-    @required ProcessManager processManager,
-    @required Logger logger,
-    @required FileSystem fileSystem,
-    @required Usage usage,
-    int majorVersion,
-    int minorVersion,
-    int patchVersion,
+    required Platform platform,
+    required ProcessManager processManager,
+    required Logger logger,
+    required FileSystem fileSystem,
+    required Usage usage,
+    int? majorVersion,
```

Comment:
What does it mean for this to have null major/minor/patch versions?

Reply:
`XcodeProjectInterpreter` populates the version in `_updateVersion()`, and the major/minor/patch properties are null if Xcode isn't installed.  In the normal constructor these aren't set, it's up to `XcodeProjectInterpreter` to maintain these values.  It's only passed into the constructor for `XcodeProjectInterpreter.test()` #74777

Reply:
instead of having multiple conditionally null fields, could this be replaced with a `Version?` getter? That would make it more difficult for xcodeproj to end up in some sort of weird state where major is non-null and minor is null et cetera

Reply:
Done.  I had to do something a little odd and make a new const `Version` initializer to make `XcodeProjectInterpreter.test()` continue to allow null (uninstalled) versions:
```dart
/// Public constant constructor when all fields are non-null, without default value fallbacks.
const Version.withText(this.major, this.minor, this.patch, this._text);
```
```dart
factory XcodeProjectInterpreter.test({
  required ProcessManager processManager,
  Version? version = const Version.withText(1000, 0, 0, '1000.0.0'),
```